quickwit/quickwit-rest-client/src/rest_client.rs

```diff
@@ -47,28 +50,41 @@ pub const DEFAULT_CLIENT_COMMIT_TIMEOUT: Timeout = Timeout::from_mins(30);
 struct Transport {
     base_url: Url,
     api_url: Url,
-    client: Client,
+    client: ClientWithMiddleware,
 }
 
 impl Transport {
-    fn new(endpoint: Url, connect_timeout: Timeout, ca_cert: Option<Certificate>) -> Self {
+    fn new(
+        endpoint: Url,
+        connect_timeout: Timeout,
+        ca_cert: Option<Certificate>,
+        num_retries: u32,
+    ) -> Self {
         let base_url = endpoint;
         let api_url = base_url
             .join("api/v1/")
-            .expect("Endpoint should not be malformed.");
-        let mut client_builder = ClientBuilder::new();
+            .expect("root url should be well-formed");
+        let mut reqwest_client_builder = ReqwestClientBuilder::new();
         if let Some(duration) = connect_timeout.as_duration_opt() {
-            client_builder = client_builder.connect_timeout(duration);
+            reqwest_client_builder = reqwest_client_builder.connect_timeout(duration);
         }
         if let Some(ca_cert) = ca_cert {
-            client_builder = client_builder
+            reqwest_client_builder = reqwest_client_builder
                 .tls_built_in_root_certs(false)
                 .add_root_certificate(ca_cert);
         }
+        let retry_policy = ExponentialBackoff::builder().build_with_max_retries(num_retries);
```

Comment:
the default max backoff seems to be 30m, that seems overly long for a cli. i'd be in favor of capping that to a few minutes

Reply:
Fair, let me tweak the retry policy.